playbooks/deploy_vms_standalone.yaml

```diff
@@ -37,54 +47,75 @@
+      when:
+        - disk_copy | bool
+        - hostvars[item].disk_extract is not defined or not hostvars[item].disk_extract | bool
+        - item not in all_vms.list_vms or (item in all_vms.list_vms and hostvars[item].force is defined and hostvars[item].force)
+      loop: "{{ groups['VMs'] }}"
+
+    - name: Copy the gzipped disk on target
       copy:
         src: "{{ hostvars[item].vm_disk }}"
         dest: "{{ disk_pool }}/{{ hostvars[item].inventory_hostname }}.img.gz"
-      with_items: "{{ groups['VMs'] }}"
-      when: disk_copy | bool and hostvars[item].disk_extract is defined and hostvars[item].disk_extract | bool
-    - name: Extract the gziped disk on target
+      when:
+        - disk_copy | bool
+        - hostvars[item].disk_extract is defined
+        - hostvars[item].disk_extract | bool
+        - item not in all_vms.list_vms or (item in all_vms.list_vms and hostvars[item].force is defined and hostvars[item].force)
+      loop: "{{ groups['VMs'] }}"
+
+    - name: Extract the gzipped disk on target
       command:
         cmd: "gzip -d -f {{ disk_pool }}/{{ hostvars[item].inventory_hostname }}.img.gz"
         creates: "{{ hostvars[item].inventory_hostname }}.img"
-      with_items: "{{ groups['VMs'] }}"
-      when: disk_copy | bool and hostvars[item].disk_extract is defined and hostvars[item].disk_extract | bool
+      when:
+        - disk_copy | bool
+        - hostvars[item].disk_extract is defined
+        - hostvars[item].disk_extract | bool
+        - item not in all_vms.list_vms or (item in all_vms.list_vms and hostvars[item].force is defined and hostvars[item].force)
+      loop: "{{ groups['VMs'] }}"
+
     - name: Add main disk to disk list
       # This is only done in standalone because the disk is handled by vm-manager in the cluster
       set_fact:
         standalone_main_disk: "{{ hostvars[item].inventory_hostname }}.qcow2"
       delegate_to: "{{ item }}"
       delegate_facts: true
-      with_items: "{{ groups['VMs'] }}"
-    - name: export VM config for debug in /tmp
+      loop: "{{ groups['VMs'] }}"
+      when: item not in all_vms.list_vms or (item in all_vms.list_vms and hostvars[item].force is defined and hostvars[item].force)
+
+    - name: Export VM config for debug in /tmp
       template:
         src: "{{ hostvars[item].vm_template }}"
         dest: "/tmp/{{ hostvars[item].inventory_hostname }}.xml"
       vars:
         vm: "{{ hostvars[item] }}"
-      with_items: "{{ groups['VMs'] }}"
       when: ansible_verbosity >= 2
+      loop: "{{ groups['VMs'] }}"
+
     - name: Create VMs
       virt:
         command: define
         xml: >-
           {{ lookup('template',
           hostvars[item].vm_template,
           template_vars=dict(vm=hostvars[item])) }}
-      with_items: "{{ groups['VMs'] }}"
+      loop: "{{ groups['VMs'] }}"
+      when: item not in all_vms.list_vms or (item in all_vms.list_vms and hostvars[item].force is defined and hostvars[item].force)
```

Comment:
I would be clearer to create an Ansible block with this the previous tasks (L51-L111)

Reply:
It is not possible to loop on a block. Solution is to create another file.

Reply:
Oh, you are right, my bad.
Ok to leave it as it is.